test-support/lit-exec-qemu.py

```diff
@@ -7,30 +7,34 @@
 # arguments as llvm-project/libcxx/utils/run.py.
 
 import sys
-import re
 import argparse
 import subprocess
 import pathlib
 
-# This script requires Python 3.6 or later
-assert sys.version_info >= (3, 6)
-
-
 def run(args):
     """Execute the program using QEMU and return the subprocess return code."""
-    command = args.command
-    assert command
-    if command[0] == '--':
-        command = command[1:]
-    image = command[0]
-    image_args = command[1:]
-    qemu_params = args.qemu_params.split(':')
-    qemu_cmd = ([args.qemu_command] + qemu_params +
-        ['-semihosting', '-nographic', '-kernel', image])
-    if image_args:
-        image_args_concat = ','.join(['arg=' + arg.replace(',', ',,')
-                                      for arg in image_args])
-        qemu_cmd += ['-semihosting-config', image_args_concat]
+    qemu_params = ["-M", args.qemu_machine]
+    if args.qemu_cpu:
+        qemu_params += ["-cpu", args.qemu_cpu]
+    if args.qemu_params:
+        qemu_params += args.qemu_params.split(':')
+    semihosting = ','.join(
+        ["enable=on", "chardev=stdio0"]
+        + ['arg=' + arg.replace(',', ',,') for arg in args.arguments])
+    qemu_params += [
```

Comment:
Comments please explaining why each argument is necessary

Reply:
Done